bitcoin/src/blockdata/transaction.rs

```diff
@@ -1970,7 +1970,7 @@ mod tests {
         let display = "0000000000000000000000000000000000000000000000000000000000000000:4294967295";
         assert_eq!(display, format!("{}", &outpoint));
 
-        let pretty_debug = "OutPoint {\n    txid: 0000000000000000000000000000000000000000000000000000000000000000,\n    vout: 4294967295,\n}";
+        let pretty_debug = "OutPoint {\n    txid: 0x0000000000000000000000000000000000000000000000000000000000000000,\n    vout: 4294967295,\n}";
```

Comment:
In 90b2ac03e3560b0e9f78f52b198c869773b5816e:

Even though this is just debug output, I think it makes the crate look sloppy and we should find a way to avoid it.

Reply:
Do you mean the change in this patch looks sloppy or that using `0x` in pretty debugout put looks sloppy? (I'm guessing the second one). Notice that this is already the current behaviour for `Txid`s, before this PR we had different behaviour for pretty printing `Txid` and `OutPoint::txid` - this PR makes it uniform.

Reply:
I mean using `0x` for txids.

Good catch finding the non-uniformity -- but we should get rid of the 0x in other places rather than adding one here.

Reply:
Can we merge this as is and fix the `Txid` as follow up? This PR really just uncovers the issue, it doesn't introduce it.

Reply:
We have https://github.com/rust-bitcoin/rust-bitcoin/issues/2498 as well.